src/app/fdctl/run/tiles/fd_shred.c

```diff
@@ -700,11 +701,13 @@ unprivileged_init( fd_topo_t *      topo,
 
   ulong expected_shred_version = tile->shred.expected_shred_version;
   if( FD_LIKELY( !expected_shred_version ) ) {
-    ulong * gossip_shred_version = (ulong*)tile->extra[0];
+    ulong busy_obj_id = fd_pod_query_ulong( topo->props, "poh_shred", ULONG_MAX );
+    FD_TEST( busy_obj_id!=ULONG_MAX );
+    ulong * gossip_shred_version = fd_fseq_join( (void*)((ulong)(topo->workspaces[ topo->objs[ busy_obj_id ].wksp_id ].wksp) + topo->objs[ busy_obj_id ].offset) );
```

Comment:
I think a macro or static inline along the lines of `fd_topo_get_laddr( topo, obj_id )` for `(void*)((ulong)(topo->workspaces[ topo->objs[ busy_obj_id ].wksp_id ].wksp) + topo->objs[ busy_obj_id ].offset)` would be worthwhile. The code to get the pointer is kind of gross.